packages/build-tools/src/ios/configure.ts

```diff
@@ -1,21 +1,49 @@
+import path from 'path';
+
 import { IOSConfig } from '@expo/config-plugins';
 import { Ios } from '@expo/eas-build-job';
+import uniq from 'lodash/uniq';
+import fs from 'fs-extra';
+import plist from '@expo/plist';
 
 import { BuildContext } from '../context';
 
 import { Credentials } from './credentials/manager';
 
-async function configureXcodeProject<TJob extends Ios.Job>(
-  ctx: BuildContext<TJob>,
+async function configureXcodeProject(
+  ctx: BuildContext<Ios.Job>,
   {
     credentials,
     buildConfiguration,
   }: {
     credentials: Credentials;
-    buildConfiguration?: string;
+    buildConfiguration: string;
   }
 ): Promise<void> {
   ctx.logger.info('Configuring Xcode project');
+  await configureCredentialsAsync(ctx, {
+    credentials,
+    buildConfiguration,
+  });
+  const { version } = ctx.job;
+  if (version?.version || version?.buildNumber) {
```

Comment:
Empty strings are also falsy.

Reply:
joi will not allow empty strings, plus event if ti was pass here we should treat them as no values, because ti would be clearly wrong

Reply:
I prefer to be explicit here.

Reply:
I don't understand what you mean by that, version at this point should not be an empty string, but if it was current behavior would be correct

Do you want me to check for `version === undefined || version === null || version === ''` or version === undefined || version === null?